build error:  error: static assertion failed: invalid specialization

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Fedora 34
- GCC 11.0.0
- Clementine version: ccba649f622497b8463930b9ad5b7a1cafe6c976

```
/builddir/build/BUILD/Clementine-ccba649f622497b8463930b9ad5b7a1cafe6c976/src/globalsearch/spotifysearchprovider.cpp:276:33:   required from here
/usr/include/c++/11/ext/numeric_traits.h:64:53: error: static assertion failed: invalid specialization
   64 |       static_assert(__is_integer_nonstrict<_Value>::__value,
      |                                                     ^~~~~~~
/usr/include/c++/11/ext/numeric_traits.h:64:53: note: 'std::__is_integer<__int128 unsigned>::__value' evaluates to false
In file included from /usr/include/c++/11/bits/stl_algo.h:66,
                 from /usr/include/c++/11/algorithm:62,
                 from /usr/include/qt5/QtCore/qglobal.h:142,
                 from /usr/include/qt5/QtCore/qiodevice.h:43,
                 from /usr/include/qt5/QtCore/qprocess.h:43,
                 from /usr/include/qt5/QtCore/QProcess:1,
                 from /builddir/build/BUILD/Clementine-ccba649f622497b8463930b9ad5b7a1cafe6c976/src/internet/spotify/spotifyservice.h:25,
                 from /builddir/build/BUILD/Clementine-ccba649f622497b8463930b9ad5b7a1cafe6c976/src/globalsearch/spotifysearchprovider.h:21,
                 from /builddir/build/BUILD/Clementine-ccba649f622497b8463930b9ad5b7a1cafe6c976/src/globalsearch/spotifysearchprovider.cpp:18:
/usr/include/c++/11/bits/uniform_int_dist.h: In instantiation of 'static _Up std::uniform_int_distribution<_IntType>::_S_nd(_Urbg&, _Up) [with _Wp = __int128 unsigned; _Urbg = std::mersenne_twister_engine<long unsigned int, 32, 624, 397, 31, 2567483615, 11, 4294967295, 7, 2636928640, 15, 4022730752, 18, 1812433253>; _Up = long unsigned int; _IntType = int]':
/usr/include/c++/11/bits/uniform_int_dist.h:306:35:   required from 'std::uniform_int_distribution<_IntType>::result_type std::uniform_int_distribution<_IntType>::operator()(_UniformRandomBitGenerator&, const std::uniform_int_distribution<_IntType>::param_type&) [with _UniformRandomBitGenerator = std::mersenne_twister_engine<long unsigned int, 32, 624, 397, 31, 2567483615, 11, 4294967295, 7, 2636928640, 15, 4022730752, 18, 1812433253>; _IntType = int; std::uniform_int_distribution<_IntType>::result_type = int]'
/usr/include/c++/11/bits/uniform_int_dist.h:192:34:   required from 'std::uniform_int_distribution<_IntType>::result_type std::uniform_int_distribution<_IntType>::operator()(_UniformRandomBitGenerator&) [with _UniformRandomBitGenerator = std::mersenne_twister_engine<long unsigned int, 32, 624, 397, 31, 2567483615, 11, 4294967295, 7, 2636928640, 15, 4022730752, 18, 1812433253>; _IntType = int; std::uniform_int_distribution<_IntType>::result_type = int]'
/builddir/build/BUILD/Clementine-ccba649f622497b8463930b9ad5b7a1cafe6c976/src/globalsearch/spotifysearchprovider.cpp:276:33:   required from here
/usr/include/c++/11/bits/uniform_int_dist.h:251:46: error: static assertion failed: W must be twice as wide as U
  251 |           static_assert(_Wp_traits::__digits == (2 * _Up_traits::__digits),
      |                                     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/11/bits/uniform_int_dist.h:251:46: note: '(((int)__gnu_cxx::__numeric_traits_integer<__int128 unsigned>::__digits) == (2 * ((int)__gnu_cxx::__numeric_traits_integer<long unsigned int>::__digits)))' evaluates to false
```

--- a/ext/clementine-spotifyblob/CMakeLists.txt
+++ b/ext/clementine-spotifyblob/CMakeLists.txt
@@ -7,7 +7,7 @@ include_directories(${CMAKE_SOURCE_DIR}/ext/libclementine-spotifyblob)
 include_directories(${CMAKE_SOURCE_DIR}/ext/libclementine-common)
 include_directories(${CMAKE_SOURCE_DIR}/src)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option --std=c++0x -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option")
 
 link_directories(${SPOTIFY_LIBRARY_DIRS})
 

--- a/ext/clementine-tagreader/CMakeLists.txt
+++ b/ext/clementine-tagreader/CMakeLists.txt
@@ -7,7 +7,7 @@ include_directories(${CMAKE_BINARY_DIR}/ext/libclementine-tagreader)
 include_directories(${CMAKE_SOURCE_DIR}/src)
 include_directories(${CMAKE_BINARY_DIR}/src)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++0x -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR})
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option --std=c++0x -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option")
 
 option(BUILD_WERROR "Build with -Werror" ON)
 

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.8.11)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -fpermissive -Wno-c++11-narrowing -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive -Wno-c++11-narrowing")
 
 if(USE_SYSTEM_GMOCK)
   include_directories(${GMOCK_INCLUDE_DIRS} ${GTEST_INCLUDE_DIRS})

